File: cpp/level1_middle_character.cpp
//
//  level1_middle_character.cpp
//  programmers_training
//
//  Created by Chanwoo Noh on 2018. 06. 20..
//  Copyright © 2018년 Chanwoo Noh. All rights reserved.
//

#include <string>
#include <vector>

using namespace std;

string solution(string s) {
    if (s.length() % 2 == 1)
        return std::string() + s.at((s.length()-1)/2);
    else
        return s.substr((int)s.length() / 2 - 1, 2);
}

File: python3/level1_middle_character.py
#
#  level1_middle_character.py
#  programmers_training
#
#  Created by Chanwoo Noh on 2018. 09. 08..
#  Copyright © 2018년 Chanwoo Noh. All rights reserved.
#

def solution(s):
    slen = len(s)
    pos = slen // 2
    if slen % 2 == 0:
        return s[pos-1:pos+1]
    else:
        return s[pos]